perfkitbenchmarker/linux_benchmarks/cloudsuite_data_caching_benchmark.py

```diff
@@ -0,0 +1,144 @@
+# Copyright 2016 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Runs the data caching benchmark of Cloudsuite 3.0.
+
+More info: http://cloudsuite.ch/datacaching
+"""
+
+import re
+
+from perfkitbenchmarker import configs
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import sample
+from perfkitbenchmarker.linux_packages import docker
+
+flags.DEFINE_string('cloudsuite_data_caching_memcached_flags',
+                    '-t 1 -m 2048 -n 550',
+                    'Flags to be given to memcached.')
+flags.DEFINE_string('cloudsuite_data_caching_rps',
+                    '18000',
+                    'Number of requests per second.')
+FLAGS = flags.FLAGS
+
+BENCHMARK_NAME = 'cloudsuite_data_caching'
+BENCHMARK_CONFIG = """
+cloudsuite_data_caching:
+  description: >
+    Runs Cloudsuite3.0 Data Caching benchmark.
+  vm_groups:
+    server:
+      vm_spec: *default_single_core
+    client:
+      vm_spec: *default_single_core
+"""
+
+
+def GetConfig(user_config):
+  return configs.LoadConfig(BENCHMARK_CONFIG, user_config, BENCHMARK_NAME)
+
+
+def Prepare(benchmark_spec):
+  """Install docker. Pull the required images from DockerHub. Create datasets.
+
+  Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+  """
+  vms = benchmark_spec.vms
+  server_vm = benchmark_spec.vm_groups['server'][0]
+  client_vm = benchmark_spec.vm_groups['client'][0]
+
+# Make sure docker is installed on all VMs.
+  for vm in vms:
+    if not docker.IsInstalled(vm):
+      vm.Install('docker')
+
+    # Prepare and start the server VM.
+  server_vm.RemoteCommand('sudo docker pull cloudsuite/data-caching:server')
+  server_vm.RemoteCommand("echo '%s    dc-client' | sudo tee -a /etc/hosts >"
+                          " /dev/null" % client_vm.internal_ip)
+  server_vm.RemoteCommand('sudo docker run --name dc-server --net host -d '
+                          'cloudsuite/data-caching:server %s' %
+                          FLAGS.cloudsuite_data_caching_memcached_flags)
+
+# Prepare the client.
+  client_vm.RemoteCommand('sudo docker pull cloudsuite/data-caching:client')
+  client_vm.RemoteCommand("echo '%s    dc-server' | sudo tee -a /etc/hosts >"
+                          " /dev/null" % server_vm.internal_ip)
+
+
+def _ParseOutput(output_str):
```

Comment:
Could this function be redone to provide more insight into what's going on? I'm worried it might be a bit flaky and difficult to debug. From the stdout I got, it looks like dc-server periodically takes measurements, and then the `numbers` assignment throws away everything except the last four lines. For reference, here's an excerpt from my output:
```
Configuration:

nProcessors on system: 8
nWorkers: 4
runtime: 123
Get fraction: 0.800000
Naggle's algorithm: False


host: dc-server
address: 10.0.0.2
Loading key value file...created uniform distribution 1000
rps 18000 cpus 4
meanInterarrival 222
num_worker_connections 50
num_worker_connections 50
Creating worker on tid 3852093184
starting receive base loop
num_worker_connections 50
Creating worker on tid 3843700480
starting receive base loop
num_worker_connections 50
Creating worker on tid 3835307776
starting receive base loop
Created 200 connections total
Creating worker on tid 3623876352
starting receive base loop
Stats:
-------------------------
  timeDiff,     rps,        requests,     gets,       sets,      hits,       misses,   avg_lat,      90th,      95th,        99th,       std,       min,        max,    avgGetSize
1461365108.363647,       0.0,       36429,      29235,       7194,      29235,          0,   0.218154,   0.264100,   0.296100,   0.389100,   0.044939,   0.000000,   1.110000, 840.211869
Outstanding requests per worker:
2 1 0 2 
  timeDiff,     rps,        requests,     gets,       sets,      hits,       misses,   avg_lat,      90th,      95th,        99th,       std,       min,        max,    avgGetSize
120.000120,   18074.9,     2168993,    1734696,     434297,    1734696,          0,   0.237822,   0.295100,   0.335100,   0.497100,   0.096340,   0.111000,  15.449000, 843.741086
Outstanding requests per worker:
1 2 1 0 
  timeDiff,     rps,        requests,     gets,       sets,      hits,       misses,   avg_lat,      90th,      95th,        99th,       std,       min,        max,    avgGetSize
120.000120,   18085.9,     2170316,    1736253,     434063,    1736253,          0,   0.240551,   0.297100,   0.339100,   0.516000,   0.149249,   0.111000,  35.241000, 843.748133
Outstanding requests per worker:
2 0 0 1 
Ran for 242.000242, exiting
```
That makes my `numbers` equal `[120.00012, 18085.9, 2170316.0, 1736253.0, 434063.0, 1736253.0, 0.0, 0.240551, 0.2971, 0.3391, 0.516, 0.149249, 0.111, 35.241, 843.748133, 2.0, 0.0, 0.0, 1.0, 242.000242]`.

Rather than using indexes and a flat array, could you parse this table to associate each value with the name of the column, and then extract the value by column name? I think that would be clearer. With the current single-regex methods, I'm worried about nan, since I see that in a previous table in my log output. I also think the outstanding requests per worker could be done as a separate step, since it doesn't fit the format of the rest of the chart.

Adding a parsing test case under tests/linux_benchmarks/cloudsuite_data_caching_benchmark_test.py would also be helpful, since it would give an example of what the output should look like.